resource/graph_node.go

```diff
@@ -11,6 +11,24 @@ import (
 	"go.viam.com/rdk/logging"
 )
 
+// NodeState captures the configuration lifecycle state of a resource node.
+type NodeState uint8
+
+const (
+	// NodeStateUnconfigured denotes a newly created resource.
+	NodeStateUnconfigured NodeState = iota
+
+	// NodeStateConfiguring denotes a resource is being configured.
+	NodeStateConfiguring
+
+	// NodeStateReady denotes a resource that has been initialized and is not being
+	// configured or removed. A resource in this state may be unhealthy.
```

Comment:
> A resource in this state may be unhealthy.

Why is there no `NodeStateUnhealthy`?

Reply:
it's a [TODO](https://github.com/viamrobotics/rdk/pull/4075/files#diff-d6f0241bd713d1ea97a4752dbb520099399b7b2edd9e94d539d0674e83cefedfR266) - I want to avoid having references to it until we actually plan to implement.